test/fn/upload-app-settings.spec.js

```diff
@@ -20,6 +20,10 @@ describe('upload-app-settings', () => {
 
     // case when testing on dev
     { coreVersion: '4.2.0-dev.1682192676689', isDeclarative: true, expectNools: false },
+    // https://github.com/medic/cht-conf/issues/596
+    { coreVersion: '4.5.0.6922454971', isDeclarative: false, expectNools: false},
```

Comment:
It was failing with version `4.5.0.6922454971`, if we pass weird build version, it'd faill.

Reply:
If I revert your changes in `src`, all these tests are still passing... This means that you have not sufficiently captured the scenario you care about in these tests. Am I mistaken?

Reply:
That's right, because the fix was on `getValidApiVersion()` and in this example, we're mocking what getValidApiVersion() returns. Previously it was failing on getValidApiVersion() and failing `upload-app-settings`. I can add unit test for `getApiVersion()` method if that makes it more useful.

Reply:
This isn't a super obvious requirement, so if you want it to not break in the future - it is typically nice to have a test which captures the requirement. Ya - that can be a unit test for `getApiVersion` or an integration test here (change what is mocked to be deeper).